src/file-events/cpp/generic_fsnotifier.cpp

```diff
@@ -163,9 +164,55 @@ void AbstractServer::reportChange(JNIEnv* env, int type, const u16string& path)
     jstring javaPath = env->NewString((jchar*) path.c_str(), (jsize) path.length());
     env->CallVoidMethod(watcherCallback, watcherCallbackMethod, type, javaPath);
     env->DeleteLocalRef(javaPath);
+
+    jthrowable exception = env->ExceptionOccurred();
+    if (exception != nullptr) {
+        env->ExceptionDescribe();
+        env->ExceptionClear();
+
+        jclass exceptionClass = env->GetObjectClass(exception);
+        assert(exceptionClass != nullptr);
+        jmethodID getMessage = env->GetMethodID(exceptionClass, "getMessage", "()Ljava/lang/String;");
```

Comment:
Do you need to delete the `getMessage` local ref as well?

Reply:
I don't think these are terribly important in any case, but the example I copied from didn't explicitly delete `jmethodID`s.

Reply:
Given that we won't throw exceptions from the callback all the time, it should be fine to leave it as is. For me, it would be less cognitive overhead if all the `j*` typed local variables would be deleted, so I don't need to remember which ones to frees and which not when reading the code.

Reply:
I just checked, and you can't pass a `jmethodID` to `DeleteLocalRef()` because it expects a `jobject`.